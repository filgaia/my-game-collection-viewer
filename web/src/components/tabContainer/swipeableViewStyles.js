const swipeableViewStyles = (theme) => ({
    root: {
        backgroundColor: theme.palette.background.paper
    }
});

export default swipeableViewStyles;
