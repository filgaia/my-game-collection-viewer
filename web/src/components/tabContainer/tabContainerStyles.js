const tabContainerStyles = (theme) => ({
    tab: {
        padding: 8 * 3
    }
});

export default tabContainerStyles;
