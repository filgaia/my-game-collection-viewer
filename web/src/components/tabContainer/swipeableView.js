// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SwipeableViews from 'react-swipeable-views';
// @material
import {
    AppBar,
    Tabs,
    Tab,
    withStyles
} from '@material-ui/core';
import {
    Favorite,
    LibraryBooks
} from '@material-ui/icons';
// @styles
import swipeableViewStyles from './swipeableViewStyles';
// @componets
import TabContainer from './tabContainer';
import Catalog from '../catalog/catalog';
// @constants
import { CATALOG_TAB, WISHLIST_TAB } from '../../constants/index';

class SwipeableView extends Component {

    componentDidMount() {
        this.props.initGames();
    }

    handleChange = (event, tab) => {
        this.props.setTab({ tab });
    };

    handleChangeIndex = tab => {
        this.props.setTab({ tab });
    };


    buildCatalogTab() {
        const { gamesInformation, loadGames, setLabelFilter } = this.props;
        const tab = gamesInformation.get('tab');
        const source = gamesInformation.get('idLabelFilter') ? 'sourceFiltered' : 'source';
        const params = {
            source: gamesInformation.get(source),
            propGames: 'games',
            propMoreItems: 'hasMoreItems'
        };

        return tab === CATALOG_TAB ? (
            <Catalog
                gamesInformation={gamesInformation}
                loadGames={(page) => loadGames(page, params)}
                hasMoreItems={gamesInformation.get('hasMoreItems')}
                games={gamesInformation.get('games')}
                setLabelFilter={setLabelFilter}
            />
        ) : null;
    }

    buildWishListTab() {
        const { gamesInformation, loadGames, setLabelFilter } = this.props;
        const tab = gamesInformation.get('tab');
        const params = {
            source: gamesInformation.get('sourceWishList'),
            propGames: 'wishList',
            propMoreItems: 'hasMoreItemsWishList'
        };

        return tab === WISHLIST_TAB ? (
            <Catalog
                gamesInformation={gamesInformation}
                loadGames={(page) => loadGames(page, params)}
                hasMoreItems={gamesInformation.get('hasMoreItemsWishList')}
                games={gamesInformation.get('wishList')}
                setLabelFilter={setLabelFilter}
            />
        ) : null;
    }


    render() {

        const { classes, theme, gamesInformation } = this.props;

        const tab = gamesInformation.get('tab');
        const catalog = this.buildCatalogTab();
        const wishList = this.buildWishListTab();
        const axis = theme.direction === 'rtl' ? 'x-reverse' : 'x';

        return (
            <div className={classes.root}>
                <AppBar position="static" color="default">
                    <Tabs
                        value={tab}
                        onChange={this.handleChange}
                        indicatorColor="primary"
                        textColor="primary"
                        variant="fullWidth"
                    >
                        <Tab label="Catalog" icon={<LibraryBooks />} />
                        <Tab label="WishList" icon={<Favorite />} />
                    </Tabs>
                </AppBar>
                <SwipeableViews
                    axis={axis}
                    index={tab}
                    onChangeIndex={this.handleChangeIndex}
                >
                    <TabContainer dir={theme.direction}>
                        {catalog}
                    </TabContainer>
                    <TabContainer dir={theme.direction}>
                        {wishList}
                    </TabContainer>
                </SwipeableViews>
            </div >
        );
    }
}

SwipeableView.propTypes = {
    classes: PropTypes.object.isRequired,
    gamesInformation: PropTypes.object.isRequired,
    initGames: PropTypes.func.isRequired,
    loadGames: PropTypes.func.isRequired,
    setLabelFilter: PropTypes.func.isRequired,
    setTab: PropTypes.func.isRequired,
    theme: PropTypes.object.isRequired
};

export default withStyles(swipeableViewStyles, { withTheme: true })(SwipeableView);
