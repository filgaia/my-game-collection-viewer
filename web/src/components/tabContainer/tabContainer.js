// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// @material
import {
    Typography,
    withStyles
} from '@material-ui/core';
// @styles
import tabContainerStyles from './tabContainerStyles';

class TabContainer extends Component {

    render() {

        const { classes, children, dir } = this.props;

        return (
            <Typography component="div" dir={dir} className={classes.tab}>
                {children}
            </Typography>
        );
    }
}

TabContainer.propTypes = {
    classes: PropTypes.object.isRequired,
    children: PropTypes.node,
    dir: PropTypes.string.isRequired
};


export default withStyles(tabContainerStyles)(TabContainer);
