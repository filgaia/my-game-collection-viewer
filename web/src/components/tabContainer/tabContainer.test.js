import React from 'react';
import renderer from 'react-test-renderer';
import { TabContainer } from './tabContainer';

describe('<TabContainer />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TabContainer
        classes={{}}
        dir="left"
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
