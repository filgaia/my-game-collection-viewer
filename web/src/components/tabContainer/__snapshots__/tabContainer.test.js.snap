// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabContainer /> renders correctly 1`] = `
<div
  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  dir="left"
/>
`;
